tests/unit/test_utils.py

```diff
@@ -106,3 +106,18 @@ def test_force_unicode(text):
     unicode_text = force_unicode(text)
     assert unicode_text == text
     assert isinstance(unicode_text, six.text_type)
+
+
+def test_force_unicode_recursion():
+    obj = {
+        'a': ['string'],
+        'b': {'str': 'yu\u3086'},
+        'c': '\x81',
+    }
+
+    clean_obj = force_unicode(obj)
+    assert isinstance(clean_obj['a'][0], six.text_type)
+    assert isinstance(clean_obj['b']['str'], six.text_type)
+    # c should fail decoding and remain unchanged
+    assert clean_obj['c'] is obj['c']
```

Comment:
I would make the unicode decoding issue a separate test as it does not really have anything to do with recursing through a list or dictionary.